src/snowcli/zipper.py

```diff
@@ -34,34 +39,24 @@ def add_file_to_existing_zip(zip_file: str, file: str):
         myzip.write(file, os.path.basename(file))
 
 
-def zip_current_dir(dest_zip: str) -> None:
-    files_to_pack = _get_list_of_files_to_pack()
-    files_to_pack = _filter_files(files_to_pack)
-    _add_files_to_zip(dest_zip, files_to_pack)
-
-
-def _get_list_of_files_to_pack() -> List[Path]:
-    return [filepath.absolute() for filepath in Path(".").glob("**/*")]
-
+def zip_dir(source: Path, dest_zip: Path) -> None:
+    files_to_pack: Iterator[Path] = filter(
+        _to_be_zipped, map(lambda f: f.absolute(), source.glob("**/*"))
+    )
 
-def _filter_files(files: List[Path]) -> List[Path]:
-    files_to_zip = []
-    for file in files:
-        file_name = file.__str__()
-
-        if file.is_dir():
-            continue
+    with ZipFile(dest_zip, "w", ZIP_DEFLATED, allowZip64=True) as package_zip:
+        for file in files_to_pack:
+            log.debug("Adding %s to %s", file, dest_zip)
+            package_zip.write(file, arcname=file.relative_to(source.absolute()))
 
-        for pattern in IGNORED_FILES:
-            if file == pattern or fnmatch.fnmatch(file_name, pattern):
-                break
-        else:
-            files_to_zip.append(file)
 
-    return files_to_zip
+def _to_be_zipped(file: Path) -> bool:
+    if file.is_dir():
+        return False
 
+    for pattern in IGNORED_FILES:
+        # This has to be a string because of fnmatch
+        if file == pattern or fnmatch.fnmatch(str(file), pattern):
```

Comment:
`file` have to be changed to string before `for`. Now first part compares `Path` and `str` and second part n times does string from `file`

Reply:
Good point, fixed